internal/role/p2p/k0s.go

```diff
@@ -0,0 +1,278 @@
+package role
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/kairos-io/kairos-sdk/machine"
+	"github.com/kairos-io/kairos-sdk/utils"
+	providerConfig "github.com/kairos-io/provider-kairos/v2/internal/provider/config"
+	service "github.com/mudler/edgevpn/api/client/service"
+)
+
+const (
+	K0sDistroName        = "k0s"
+	K0sMasterName        = "controller"
+	K0sWorkerName        = "worker"
+	K0sMasterServiceName = "k0scontroller"
+	K0sWorkerServiceName = "k0sworker"
+)
+
+type K0sNode struct {
+	providerConfig *providerConfig.Config
+	roleConfig     *service.RoleConfig
+	ip             string
+	role           string
+}
+
+func (k *K0sNode) IsWorker() bool {
+	return k.role == RoleWorker
+}
+
+func (k *K0sNode) K8sBin() string {
+	return utils.K0sBin()
+}
+
+func (k *K0sNode) DeployKubeVIP() error {
+	pconfig := k.ProviderConfig()
+	if pconfig.KubeVIP.IsEnabled() {
+		return errors.New("KubeVIP is not yet supported with k0s")
+	}
+
+	return nil
+}
+
+func (k *K0sNode) GenArgs() ([]string, error) {
+	var args []string
+
+	// Generate a new k0s config
+	_, err := utils.SH("k0s config create > /etc/k0s/k0s.yaml")
+	if err != nil {
+		return args, err
+	}
+	args = append(args, "--config /etc/k0s/k0s.yaml")
+
+	// set the IP address of the node
+	_, err = utils.SH(fmt.Sprintf("sed -i 's/address: .*/address: %s/' /etc/k0s/k0s.yaml", k.IP()))
```

Comment:
I think it's going to be better to read the file and change the values instead of doing this wild wild west change here 😅

Reply:
reading the yaml fiel and changing the values directly